tests/performance/local/libcds_hp_overhead.cpp

```diff
@@ -0,0 +1,344 @@
+
+#include <hpx/include/parallel_execution.hpp>
+#include <hpx/modules/synchronization.hpp>
+#include <hpx/thread_executors/limiting_executor.hpp>
+
+#include <cds/gc/hp.h>    // for cds::HP (Hazard Pointer) SMR
+#include <cds/init.h>     // for cds::Initialize and cds::Terminate
+
+#include <array>
+#include <atomic>
+#include <cstddef>
+#include <cstdint>
+#include <iostream>
+#include <sstream>
+#include <stdexcept>
+#include <string>
+#include <type_traits>
+#include <vector>
+
+using hpx::program_options::options_description;
+using hpx::program_options::value;
+using hpx::program_options::variables_map;
+
+using hpx::finalize;
+using hpx::init;
+
+using hpx::find_here;
+using hpx::naming::id_type;
+
+using hpx::apply;
+using hpx::async;
+using hpx::future;
+using hpx::lcos::wait_each;
+
+using hpx::util::high_resolution_timer;
+
+using hpx::cout;
+using hpx::flush;
+
+// global vars we stick here to make printouts easy for plotting
+static std::string queuing = "default";
+static std::size_t numa_sensitive = 0;
+static std::uint64_t num_threads = 1;
+static std::string info_string = "";
+
+///////////////////////////////////////////////////////////////////////////////
+void print_stats(const char* title, const char* wait, const char* exec,
+    std::int64_t count, double duration, bool csv, bool libcds)
+{
+    std::ostringstream temp;
+    double us = 1e6 * duration / count;
+    if (csv)
+    {
+        hpx::util::format_to(temp,
+            "{1}, {:27}, {:15}, {:28}, {:8}, {:8}, {:20}, {:4}, {:4}, "
+            "{:20}, {:4}",
+            count, title, wait, exec, duration, us, queuing, numa_sensitive,
+            num_threads, info_string, libcds);
+    }
+    else
+    {
+        hpx::util::format_to(temp,
+            "invoked {:1} futures, {:27} {:15} {:28} in {:8} seconds : {:8} "
+            "us/future, queue {:20}, numa {:4}, threads {:4}, info {:20}"
+            ", libcds {:4}",
+            count, title, wait, exec, duration, us, queuing, numa_sensitive,
+            num_threads, info_string, libcds);
+    }
+    std::cout << temp.str() << std::endl;
+    // CDash graph plotting
+    //hpx::util::print_cdash_timing(title, duration);
+}
+
+const char* exec_name(hpx::parallel::execution::parallel_executor const& exec)
+{
+    return "parallel_executor";
+}
+
+const char* exec_name(
+    hpx::parallel::execution::parallel_executor_aggregated const& exec)
+{
+    return "parallel_executor_aggregated";
+}
+
+const char* exec_name(
+    hpx::parallel::execution::thread_pool_executor const& exec)
+{
+    return "thread_pool_executor";
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// we use globals here to prevent the delay from being optimized away
+double global_scratch = 0;
+std::uint64_t num_iterations = 0;
+
+///////////////////////////////////////////////////////////////////////////////
+double null_function(bool uselibcds) noexcept
+{
+#ifdef CDS_THREADING_HPX
```

Comment:
If libCDS is not configured to support HPX threads this test should not be enabled in the first place.